# Ejemplo 5.1 de Johnson & Wichern (2007)
<img src="https://raw.githubusercontent.com/fhernanb/fhernanb.github.io/master/my_docs/logo_unal_color.png" alt="drawing" width="200"/>

En este ejemplo se desea estudiar el siguiente conjunto de hipótesis usando un nivel de significancia del 0.05 y un conjunto de datos pequeño (3 observaciones).

$H_0: \boldsymbol{\mu} = (9, 5)^\top$

$H_A: \boldsymbol{\mu} \neq (9, 5)^\top$

In [1]:
import numpy as np
import pandas as pd
import scipy.stats

A continuación los datos del ejemplo

In [2]:
dt = {'x1': [6, 10, 8], 'x2': [9, 6, 3]}
dt = pd.DataFrame(data=dt)
dt

x1  x2
0   6   9
1  10   6
2   8   3

<div class="alert alert-danger">
  <strong>Cuidado!</strong> Se debe verificar que la muestra aleatoria provenga de una población normal multivariada. Aquí vamos a asumir que este supuesto se cumple.
</div>

El vector de medias se obtiene así:

In [3]:
x_barra = dt.mean()
x_barra = np.matrix(x_barra).transpose()  # Para convertir a matriz columna
print(x_barra)                            # Para ver el contenido

[[8.]
 [6.]]


La matriz de varianzas se obtiene así:

In [4]:
S = dt.cov()
S = np.matrix(S) # Para convertir a matriz
S

matrix([[ 4., -3.],
        [-3.,  9.]])

El vector de referencia es $\boldsymbol{\mu}_0=(9, 5)^\top$.

In [5]:
mu0 = [[9], [5]]
mu0

[[9], [5]]

Ahora calculamos $T^2=n (\bar{\boldsymbol{x}}-\boldsymbol{\mu}_0)^\top \boldsymbol{S}^{-1} (\bar{\boldsymbol{x}}-\boldsymbol{\mu}_0)$, la cantidad $(\bar{\boldsymbol{x}}-\boldsymbol{\mu}_0)$ se va a llamar `aux` para facilitar las operaciones.

In [6]:
n = 3 # número de observaciones
aux = np.matrix((x_barra - mu0)) # Elemento auxiliar
T2 = n * np.matmul(np.matmul(aux.transpose(), S.I), aux)
T2.item()  # Para obtener el T2 como un escalar y no como una matriz

0.7777777777777777

El valor crítico para realizar la prueba es $\frac{(n-1)p}{(n-p)} F_{p, n-p}(\alpha)$. Para obtener el valor crítico con $\alpha=0.05$ se procede así:

In [7]:
dfn = 2      # Grados de libertad del numerador
dfd = 3 - 2  # Grados de libertad del denominador
(3-1) * 2 * scipy.stats.f.ppf(q=1-0.05, dfn=dfn, dfd=dfd) / (3-2)   # Valor crítico

797.9999999999989

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $T^2=0.78$, que es menor que el valor crítico $798$, entonces no hay evidencias suficientes para rechazar $H_0: \boldsymbol{\mu} = (9, 5)^\top$.
</div>

<div class="alert alert-info">
  <strong>Tarea</strong> Replicar el ejemplo 5.2 de Johnson & Wichern (2007).
</div>